projects/typesystem/projects/common/src/data_path.rb

```diff
@@ -5,7 +5,7 @@ class DataPath
     class BadPathError < StandardError; end
 
     # TODO: use this wherever it makes sense
-    EMPTY_PATH = [].freeze
+    EMPTY_PATH = EMPTY_ARRAY
```

Comment:
nice, also perfectly fine to eliminate `EMPTY_PATH` if you want and just call `EMPTY_ARRAY` in those spots. Just depends. One has fewer constants and one has somewhat more self-documenting naming. Either is fine with me.